### ISSUE TITLE
能不能支持gpu 推理

### ISSUE THREAD

**Issue by shuishu** [mozillazg/pypinyin-g2pW#4]
能不能支持gpu 推理

**Comment by bensonbs** [comment]
**調用 GPU 推理功能**

測試以下作法能夠成功調用GPU推理

使用 onnxruntime gpu版本
```
pip install onnxruntime-gpu
```

測試是否安裝成功
```
import onnxruntime
print(onnxruntime.get_device())
GPU
```
測試是否調用成功
```
ort_session = onnxruntime.InferenceSession("./G2PWModel-v2-onnx/g2pw.onnx",providers=['CUDAExecutionProvider'])
print(ort_session.get_providers())
['CUDAExecutionProvider', 'CPUExecutionProvider']
```

理論上只須修改此行
https://github.com/GitYCC/g2pW/blob/36b02d4d7058160cef8112a458aa2cdcd1c8e5bf/g2pw/api.py#L70

```
self.session_g2pw =  onnxruntime.InferenceSession(os.path.join(model_dir, 'g2pw.onnx'), sess_options=sess_options,providers=['CUDAExecutionProvider'])
```

**Comment by ZhikangNiu** [comment]
> **調用 GPU 推理功能**
> 
> 測試以下作法能夠成功調用GPU推理
> 
> 使用 onnxruntime gpu版本
> 
> ```
> pip install onnxruntime-gpu
> ```
> 
> 測試是否安裝成功
> 
> ```
> import onnxruntime
> print(onnxruntime.get_device())
> GPU
> ```
> 
> 測試是否調用成功
> 
> ```
> ort_session = onnxruntime.InferenceSession("./G2PWModel-v2-onnx/g2pw.onnx",providers=['CUDAExecutionProvider'])
> print(ort_session.get_providers())
> ['CUDAExecutionProvider', 'CPUExecutionProvider']
> ```
> 
> 理論上只須修改此行 https://github.com/GitYCC/g2pW/blob/36b02d4d7058160cef8112a458aa2cdcd1c8e5bf/g2pw/api.py#L70
> 
> ```
> self.session_g2pw =  onnxruntime.InferenceSession(os.path.join(model_dir, 'g2pw.onnx'), sess_options=sess_options,providers=['CUDAExecutionProvider'])
> ```

Yeah, it couldn't work on onnxruntime-gpu=1.18.0 / 1.18.1 on CUDA 12.4

**Comment by FeelCoolOne** [comment]
> **調用 GPU 推理功能**
> 
> 測試以下作法能夠成功調用GPU推理
> 
> 使用 onnxruntime gpu版本
> 
> ```
> pip install onnxruntime-gpu
> ```
> 
> 測試是否安裝成功
> 
> ```
> import onnxruntime
> print(onnxruntime.get_device())
> GPU
> ```
> 
> 測試是否調用成功
> 
> ```
> ort_session = onnxruntime.InferenceSession("./G2PWModel-v2-onnx/g2pw.onnx",providers=['CUDAExecutionProvider'])
> print(ort_session.get_providers())
> ['CUDAExecutionProvider', 'CPUExecutionProvider']
> ```
> 
> 理論上只須修改此行 https://github.com/GitYCC/g2pW/blob/36b02d4d7058160cef8112a458aa2cdcd1c8e5bf/g2pw/api.py#L70
> 
> ```
> self.session_g2pw =  onnxruntime.InferenceSession(os.path.join(model_dir, 'g2pw.onnx'), sess_options=sess_options,providers=['CUDAExecutionProvider'])
> ```

me too, Dnot work on onnxruntime-gpu=1.19.2 and CUDA 12.2

**Comment by H722X** [comment]
work on onnxruntime-gpu=1.19.2 and CUDA 12.6
